parser/fsimporter.go

```diff
@@ -58,22 +52,16 @@ type (
 )
 
 //NewFSImporter creates a new file system importer
-func NewFSImporter(res *resources.Resources,
-	indexingThreads int, parseThreads int, importFiles []string) *FSImporter {
+func NewFSImporter(res *resources.Resources) *FSImporter {
+	// set batchSize to the max of 4GB or a half of system RAM to prevent running out of memory while importing
+	batchSize := int64(util.MaxUint64(4*(1<<30), (memory.TotalMemory() / 2)))
```

Comment:
In 1cf8e14, I changed the batching limit to take the greater of the old default (4GB) or half of the system's total memory. This allows us to run the MongoDB analysis phase quicker when the system can support it.